carcara/src/ast/printer.rs

```diff
@@ -315,24 +315,15 @@ fn escape_string(string: &str) -> Cow<str> {
 
 impl fmt::Display for Term {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        match self {
-            Term::Terminal(t) => write!(f, "{}", t),
-            Term::App(func, args) => write_s_expr(f, func, args),
-            Term::Op(op, args) => write_s_expr(f, op, args),
-            Term::Sort(sort) => write!(f, "{}", sort),
-            Term::Quant(quantifier, bindings, term) => {
-                write!(f, "({} {} {})", quantifier, bindings, term)
-            }
-            Term::Choice((symbol, sort), term) => {
-                write!(f, "(choice (({} {})) {})", symbol, sort, term)
-            }
-            Term::Let(bindings, term) => {
-                write!(f, "(let {} {})", bindings, term)
-            }
-            Term::Lambda(bindings, term) => {
-                write!(f, "(lambda {} {})", bindings, term)
-            }
-        }
+        let mut buf = Vec::new();
+        let mut printer = AlethePrinter {
+            inner: &mut buf,
+            term_indices: Some(AHashMap::new()),
+            term_sharing_variable_prefix: "@p_",
+        };
+        printer.write_raw_term(&self).unwrap();
```

Comment:
One very tiny thing: here you don't need to borrow `self` since it is already a `&Term`. You can just do `printer.write_raw_term(self)`. (this was caught by `cargo clippy`)